src/rf/apps/workers/status_updates.py

```diff
@@ -10,49 +10,234 @@
 import apps.core.enums as enums
 
 ERROR_MESSAGE_LAYER_IMAGE_INVALID = 'Cannot process invalid images.'
+ERROR_MESSAGE_LAYER_IMAGE_TRANSFER = 'Processing aborted. Missing images.'
 
 
-def mark_image_valid(s3_uuid):
+def mark_image_uploaded(s3_uuid):
     try:
         image = LayerImage.objects.get(s3_uuid=s3_uuid)
     except LayerImage.DoesNotExist:
         return False
 
-    image.status = enums.STATUS_VALID
+    image = update_layerimage_status_start(image, enums.STATUS_UPLOAD)
+    image = update_layerimage_status_end(image, enums.STATUS_UPLOAD)
     image.save()
+
+    # It is okay to repeat this. It will not overwrite existing timestamps.
+    if not mark_layer_status_start(image.layer_id, enums.STATUS_UPLOAD):
+        return False
```

Comment:
I think status updates should be atomic. If the status can't update for some reason, an exception should be raised. This should remove these kind of logic branches which shouldn't be necessary.